libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -1037,10 +1043,12 @@ IotMqttError_t IotMqtt_Init( void )
 
 void IotMqtt_Cleanup( void )
 {
-    if( _initCalled == 1 )
-    {
-        _initCalled = 0;
+    uint32_t initializationStatus = Atomic_CompareAndSwap_u32( &_initCalled,
+                                                               MQTT_LIBRARY_UNINITIALIZED,
+                                                               MQTT_LIBRARY_INITIALIZED );
 
+    if( initializationStatus == MQTT_LIBRARY_UNINITIALIZED )
```

Comment:
Is this correct? Since `Atomic_CompareAndSwap_u32` returns `1` when successful, if `initializationStatus` were to equal `MQTT_LIBRARY_UNINITIALIZED`, won't that mean the swap failed, and `_initCalled` was NOT equal to `MQTT_LIBRARY_INITIALIZED` beforehand?

Reply:
Good catch, that should be comparing `!= 0` just like the init case.